### ISSUE TITLE
An interface with IP aliasing is not working in MAC OS X

### ISSUE THREAD

**Issue by acetcom** [sctplab/usrsctp#178]
My Macbook-Air has one interface as follows.
```
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 84:38:35:42:17:0e 
	inet6 fe80::8638:35ff:fe42:170e%en0 prefixlen 64 scopeid 0x4 
	inet 192.168.0.6 netmask 0xffffff00 broadcast 192.168.0.255
	nd6 options=1<PERFORMNUD>
	media: autoselect
	status: active
```

In this situation, SCTP is normally working using provided example.
```
sudo ./tsctp -L 192.168.0.6
sudo ./client 192.168.0.6 5001
```

However, if I add one IP alias to `en0` interface, it is not working.
```
sudo ifconfig en0 alias 192.168.0.100 netmask 255.255.255.255 up
```

In this case, any sample example is not working.
```
sudo ./tsctp -L 192.168.0.6
sudo ./client 192.168.0.6 5001
```
and
```
sudo ./tsctp -L 192.168.0.100
sudo ./client 192.168.0.100 5001
```

Is this problem related to NAT-box involved issue?

**Comment by tuexen** [comment]
No, it is related to the fact that binding addresses in the userland stack doesn't work properly. It is easy to do in kernel land, but not that easy in userland...

**Comment by acetcom** [comment]
No Problem! I'd just like to know the reason. My project has some method to avoid this situations.

For example, some individual want to use the internal communication with IP aliasing. In that case, they can create these IP address in the loopback interface.
> sudo ifconfig lo0 alias 10.1.35.214 netmask 255.255.255.255
> sudo ifconfig lo0 alias 10.1.35.215 netmask 255.255.255.255
> ...

Any sample example is working properly if `en0` interface have only one IP address.
> sudo ./tsctp -L 192.168.0.6
> sudo ./client 192.168.0.6 5001
